bin/c8.js

```diff
@@ -1,35 +1,12 @@
 #!/usr/bin/env node
 'use strict'
 
-const fs = require('fs')
-const foreground = require('foreground-child')
-const { outputReport } = require('../lib/commands/report')
-const { promisify } = require('util')
-const rimraf = require('rimraf')
-const {
-  buildYargs,
-  hideInstrumenteeArgs,
-  hideInstrumenterArgs
-} = require('../lib/parse-args')
-
-const instrumenterArgs = hideInstrumenteeArgs()
-let argv = buildYargs().parse(instrumenterArgs)
-
-;(async function run () {
-  if ([
-    'check-coverage', 'report'
-  ].indexOf(argv._[0]) !== -1) {
-    argv = buildYargs(true).parse(process.argv.slice(2))
-  } else {
-    if (argv.clean) {
-      await promisify(rimraf)(argv.tempDirectory)
-      await promisify(fs.mkdir)(argv.tempDirectory, { recursive: true })
-    }
-
-    process.env.NODE_V8_COVERAGE = argv.tempDirectory
-    foreground(hideInstrumenterArgs(argv), async (done) => {
-      await outputReport(argv)
-      done()
-    })
-  }
-})()
+const minNode = `10.12.0`
+const semver = require('semver')
+
+/* c8 ignore next 3 */
+if (semver.lt(process.version, minNode)) {
+  console.warn(`c8 requires Node v${minNode} or greater, found ${process.version}`)
```

Comment:
Users expect `c8` to exit with non-zero code when they see this message.

Reply:
because coverage is an optional part of the testing process, I'd rather exit with `0`; this allows a user to run coverage for their entire testing matrix on a system like Travis:

`node 8`, `node 10, `node 11`, as an example, and they won't need special logic to prevent coverage from running on `node 8` (it just won't output anything) -- I actually have this use-case, which is why I added this feature.

Reply:
seem reasonable for now? would be happy to revisit in the future if this approach doesn't work well for people.

Reply:
OK, I agree on exiting with `0`. Thanks for the explanation.

> they won't need special logic to prevent coverage from running on `node 8` (it just won't output anything) 

Shouldn't the passed command, for example `tap` when they run `c8 tap`, be executed without coverage reporting, in this case?

Reply:
(@bcoe already posted https://github.com/bcoe/c8/pull/88#issuecomment-488570240 while I'm writing https://github.com/bcoe/c8/pull/88#discussion_r280305528)